### ISSUE TITLE
Structure-oriented Encoder

### ISSUE THREAD

**Issue by xinsheng44** [working-yuhao/DEAL#4]
hello,the formula of structure encoder in this paper is shown in the figure below.

![图片](https://user-images.githubusercontent.com/59272013/109760719-28ce2000-7c2a-11eb-91e5-2586a5c0e99f.png)

but I read the code, and it is true that the node is directly evaluated nn.embedding.
I didn't find the code written according to this formula

`
    def node_forward(self, nodes):
        #self.node_emb = nn.Embedding(node_num, emb_dim).to(self.device)
        first_embs = self.node_emb(nodes[:,0])
        sec_embs = self.node_emb(nodes[:,1])
        return self.node_layer(first_embs,sec_embs)
`
Is it because the published code is incomplete or I didn't find it? Could you help me find it? i will thank you very much.

**Comment by working-yuhao** [comment]
Hi,

Sorry for the delay on this one.
Yes, you are right. The released code is a simple version that doesn't include the weight normalization. 
To use it, please refer to https://pytorch.org/docs/stable/generated/torch.nn.utils.weight_norm.html.
Actually, the structure-oriented Encoder can be replaced by any other neural network, and we here just propose the simplest example for testing.

kind regards,
Yu
